male-seed-beetle/scripts/convert-with-core.py

```diff
@@ -1,7 +1,57 @@
 import polars as pl
+import re
+from pathlib import Path
+
+#Transform from tab sep to comma sep
+with open('../data-raw/data.csv', 'r') as tsv:
+
+    with open('../data-raw/data1.csv', 'w') as csv:
+
+        for line in tsv:
+
+            content = re.sub("\t", ",", line) 
+
+            csv.write(content)
+
+Path('../data-raw/data.csv').unlink()
+
+Path('../data-raw/data1.csv').rename('../data-raw/data.csv')
 
 df = pl.read_csv('../data-raw/data.csv')
 
-#Test that it works - remove later!
-first_row = df.head(1)
-print(first_row)
+#Convert categorical values to characters
+
+block_mapping = {1:"Block 1", 2:"Block 2", 3:"Block 3"}
+
+df = df.with_columns(
+    BLOCK = pl.col("BLOCK").replace_strict(block_mapping, return_dtype=pl.Utf8)
+)
```

Comment:
```suggestion
df = df.rename({col: col.lower() for col in df.columns})
    .with_columns(
        block=pl.col("block").replace_strict(block_mapping, return_dtype=pl.Utf8),
        treatment=pl.col("treatment").replace_strict(treatment_map),
    )
```

I think something like this will work, called method chaining.

Plus you can modify or add multiple columns within `with_columns()`.